include/alpaka/dev/DevOacc.hpp

```diff
@@ -0,0 +1,343 @@
+        {}
+    public:
+        //-----------------------------------------------------------------------------
+        DevOacc(DevOacc const &) = default;
+        //-----------------------------------------------------------------------------
+        DevOacc(DevOacc &&) = default;
+        //-----------------------------------------------------------------------------
+        auto operator=(DevOacc const &) -> DevOacc & = default;
+        //-----------------------------------------------------------------------------
+        auto operator=(DevOacc &&) -> DevOacc & = default;
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST auto operator==(DevOacc const & rhs) const
+        -> bool
+        {
+            return m_spDevOaccImpl->iDevice() == rhs.m_spDevOaccImpl->iDevice();
+        }
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST auto operator!=(DevOacc const & rhs) const
+        -> bool
+        {
+            return !((*this) == rhs);
+        }
+        //-----------------------------------------------------------------------------
+        ~DevOacc() = default;
+        int iDevice() const {return m_spDevOaccImpl->iDevice();}
+        acc_device_t deviceType() const {return m_spDevOaccImpl->deviceType();}
+        void makeCurrent() const
+        {
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+            std::cout << "acc_set_device_num( " << m_spDevOaccImpl->iDevice() << ", [type] )" << std::endl;
+#endif
+            acc_set_device_num(m_spDevOaccImpl->iDevice(), m_spDevOaccImpl->deviceType());
+        }
+
+        ALPAKA_FN_HOST auto getAllQueues() const
+        -> std::vector<std::shared_ptr<IGenericThreadsQueue<DevOacc>>>
+        {
+            return m_spDevOaccImpl->getAllExistingQueues();
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Registers the given queue on this device.
+        //! NOTE: Every queue has to be registered for correct functionality of device wait operations!
+        ALPAKA_FN_HOST auto registerQueue(std::shared_ptr<IGenericThreadsQueue<DevOacc>> spQueue) const
+        -> void
+        {
+            m_spDevOaccImpl->registerQueue(spQueue);
+        }
+
+    public:
+        std::shared_ptr<oacc::detail::DevOaccImpl> m_spDevOaccImpl;
+    };
+
+    namespace traits
+    {
+        //#############################################################################
+        //! The OpenACC device name get trait specialization.
+        template<>
+        struct GetName<
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST static auto getName(
+                DevOacc const &)
+            -> std::string
+            {
+                return std::string("OpenACC target");
```

Comment:
The extra construction of a string is not necessary. `return "OpenACC target"` should suffice.

Reply:
@jkelling

Reply:
I know, that there is an implicit cast, but alpaka likes to be very explicit. This is done in the same way in e.g. the CUDA backend: https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/dev/DevUniformCudaHipRt.hpp#L122